packages/memory-storage/src/resource-clients/key-value-store.ts

```diff
@@ -151,23 +157,25 @@ export class KeyValueStoreClient extends BaseClient {
             return a.key.localeCompare(b.key);
         });
 
-        let truncatedItems = items;
+        const filteredItems = items.filter((item) => !prefix || item.key.startsWith(prefix));
+
+        let truncatedItems = filteredItems;
         if (exclusiveStartKey) {
-            const keyPos = items.findIndex((item) => item.key === exclusiveStartKey);
-            if (keyPos !== -1) truncatedItems = items.slice(keyPos + 1);
+            const keyPos = filteredItems.findIndex((item) => item.key === exclusiveStartKey);
+            if (keyPos !== -1) truncatedItems = filteredItems.slice(keyPos + 1);
         }
 
         const limitedItems = truncatedItems.slice(0, limit);
 
-        const lastItemInStore = items[items.length - 1];
+        const lastItemInStore = filteredItems[filteredItems.length - 1];
```

Comment:
nit - you can use `filteredItems.at(-1)` these days

Reply:
strictly speaking, we support node 16.0+ and this was introduced in 16.6, but i guess we can ignore pre-lts versions of node 16 nowadays